mm/ktsan/stack_depot.c

```diff
@@ -0,0 +1,132 @@
+#include "ktsan.h"
+
+#include <linux/jhash.h>
+#include <linux/kernel.h>
+
+#define KT_STACK_HASH_SEED 0x9747b28c
+
+/* Only available during early boot. */
+void __init kt_stack_depot_init(kt_stack_depot_t *depot,
+			unsigned part_count, unsigned stack_count)
+{
+	unsigned i;
+
+	depot->part_count = part_count;
+	depot->stack_count = stack_count;
+
+	kt_cache_init(&depot->part_cache,
+			sizeof(*depot->parts) * part_count, 1);
+	depot->parts = kt_cache_alloc(&depot->part_cache);
+	BUG_ON(depot->parts == NULL);
+
+	for (i = 0; i < part_count; i++) {
+		depot->parts[i].head = NULL;
+		depot->parts[i].hash = 0;
+	}
+
+	kt_cache_init(&depot->stack_cache,
+		sizeof(kt_stack_depot_obj_t), stack_count);
+
+	kt_spin_init(&depot->lock);
+}
+
+/* Only available during early boot. */
+void __init kt_stack_depot_destroy(kt_stack_depot_t *depot)
+{
+	kt_cache_destroy(&depot->part_cache);
+	kt_cache_destroy(&depot->stack_cache);
+	depot->parts = NULL;
+}
+
+static inline u32 kt_stack_hash(kt_stack_t *stack)
+{
+	return jhash2((u32 *)stack->pc,
+			stack->size * sizeof(u32) / sizeof(stack->pc[0]),
+			KT_STACK_HASH_SEED);
+}
+
+static inline bool kt_stack_compare(kt_stack_t *first, kt_stack_t *second)
+{
+	if (first->size != second->size)
+		return false;
+	return memcmp(&first->pc, &second->pc,
+			sizeof(first->pc[0]) * first->size);
+}
+
+
+static inline kt_stack_depot_obj_t *kt_stack_depot_lookup_obj(
```

Comment:
depot is unused here
I would delete it as it is file-scope function, if we need to add more arguments it will be easy to do